src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -91,6 +93,79 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
             return args.Count > 0 && args[0] is PythonCollection c ? c.Contents.FirstOrDefault() : null;
         }
 
+        public static IMember Super(IPythonModule declaringModule, IPythonFunctionOverload overload, IArgumentSet argSet, IndexSpan indexSpan) {
+            if (argSet.Arguments.Any()) {
+                // If multiple arguments first argument is required
+                var classType = argSet.Argument<IMember>(0).GetPythonType<IPythonClassType>();
+                if (classType?.Mro == null) {
+                    return null;
+                }
+
+                // second argument optional
+                bool isUnbound = argSet.Argument<IMember>(1) == null;
+                if (isUnbound) {
+                    return CreateSuper(argSet, classType.Mro);
+                } else {
+                    // Second argument can be either an object instance or a class which are both IPythonType
+                    IPythonType objOrType = null;
+                    objOrType = argSet.Argument<IMember>(1).GetPythonType();
+                    bool isInstance = objOrType?.Equals(classType) ?? false;
+
+                    if (!isInstance) {
+                        var argAsType = argSet.Argument<IMember>(1).GetPythonType<IPythonClassType>();
+                        bool isSubClass = classType?.Bases.Any(b => argAsType.Equals(b)) ?? false;
```

Comment:
argAsType can be null so you may end up calling `Equals` on null. Either check or reverse comparison like `b.Equals(argAsType)`. Also, `Bases` can be null. Feel free to write helper like `IsSuperClassOf(this IPythonClassType, ...)`.

Reply:
Also, I think if class belongs to bases it is a `superClass`. Subclass is whatever is derived.

Reply:
Good catch. ya I found that null case when testing.

I'm actually looking for derived, so I made IsSubClassOf(..), I re-read the python3 docs and I was traversing the wrong mro.

Reply:
I went with IsSubClassOf because the docs mention object-type(param2) being a subclass of type(param1)